<launch>
<!-- Launch file for Livox AVIA LiDAR -->

	<arg name="rviz" default="true" />

	<rosparam command="load" file="$(find fast_lio)/config/avia.yaml" />

	<param name="feature_extract_enable" type="bool" value="0"/>
	<param name="point_filter_num" type="int" value="1"/>
	<param name="max_iteration" type="int" value="3" />
	<param name="filter_size_surf" type="double" value="0.5" />
	<param name="filter_size_map" type="double" value="0.5" />
	<param name="cube_side_length" type="double" value="1000" />
	<param name="runtime_pos_log_enable" type="bool" value="0" />
	<param name="pcd_save_enable" type="bool" value="0" />
    <node pkg="fast_lio" type="fastlio_mapping" name="laserMapping" output="screen" />

	<arg name="x" default="14.5" />
    <arg name="y" default="-7.5" />
    <arg name="z" default="0.0" />
    <arg name="roll" default="-0.25" />
    <arg name="pitch" default="0.0" />
    <arg name="yaw" default="0.0" />

	<arg name="user" default="$(env USER)" />
	<arg name="map" default="/home/$(arg user)/catkin_ws/pcd/test_map.pcd" />

    <!-- loalization-->
    <node pkg="fast_lio" type="global_localization.py" name="global_localization" output="screen">
      <param name="x" value="$(arg x)" />
      <param name="y" value="$(arg y)" />
      <param name="z" value="$(arg z)" />
      <param name="roll" value="$(arg roll)" />
      <param name="pitch" value="$(arg pitch)" />
      <param name="yaw" value="$(arg yaw)" />
    </node>

	<!-- transform  fusion-->
    <node pkg="fast_lio" type="transform_fusion.py" name="transform_fusion" output="screen" />

	<!-- glbal map-->
    <node pkg="pcl_ros" type="pcd_to_pointcloud" name="map_publishe" output="screen"
		  args="$(arg map) 5 _frame_id:=map cloud_pcd:=/map" />


	<group if="$(arg rviz)">
	<node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find fast_lio)/rviz_cfg/localization.rviz" />
	</group>

</launch>
